docs/share_usage.md

```diff
@@ -0,0 +1,611 @@
+      storage: 10Gi
+    accessModes:
+    - ReadWriteMany
+    flexVolume:
+      driver: hpe.com/hpe
+      options: 
+        filePersona: ""
+EOF
+```
+
+Persistent Volume Claim Example
+Now let’s create a claim PersistentVolumeClaim (PVC). Here we specify the PVC name pvc-first.
+
+```yaml
+$ sudo kubectl create -f - << EOF
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: pvc-first
+spec:
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 10Gi
+EOF
+```
+
+At this point, after creating the PV and PVC definitions, the volume hasn’t been created yet. The actual volume gets created on-the-fly during the pod deployment and volume mount phase.
+
+Pod Example
+So, let’s create a pod "pod-first" using the minio container along with some persistent storage:
+
+```yaml
+$ sudo kubectl create -f - << EOF
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-first
+spec:
+  containers:
+  - name: minio
+    image: minio/minio:latest
+    args:
+    - server
+    - /export
+    env:
+    - name: MINIO_ACCESS_KEY
+      value: minio
+    - name: MINIO_SECRET_KEY
+      value: doryspeakswhale
+    ports:
+    - containerPort: 9000
+    volumeMounts:
+    - name: export
+      mountPath: /export
+  volumes:
+    - name: export
+      persistentVolumeClaim:
+        claimName: pvc-first
+EOF
+```
+Now the **pod** can be deleted to unmount the Docker volume. Deleting a **Docker volume** does not require manual clean-up because the dynamic provisioner provides automatic clean-up. You can delete the **PersistentVolumeClaim** and see the **PersistentVolume** and **Docker volume** automatically deleted.
+
+
+Congratulations, you have completed all validation steps and have a working **Kubernetes/OpenShift** environment.
+
+### Restarting the Containerized plugin<a name="restart"></a>
+
+If you need to restart the containerized plugin used in Kubernetes/OpenShift environments, run the following command:
+
+```
+$ docker stop <container_id_of_plugin>
+```
+
+>Note: The /run/docker/plugins/hpe.sock and /run/docker/plugins/hpe.sock.lock files are not automatically removed when you stop the container. Therefore, these files will need to be removed manually between each run of the plugin.
+
+```
+$ docker start <container_id_of_plugin>
+```
+
+## Limitations / Known Issues
+1. All the operations must be performed sequentially. E.g. concurrent creation 
+   of multiple shares can lead to ETCD lock failures.
+2. When block related configuration parameters are used inadvertently in file 
+   driver configuration or vice-versa, it does not result in any error - the
+   plugin simply ignores it. Eg: snapcpg, a block configuration parameter, 
+   when used in file driver configuration, it is ignored.
+3. When two backend sections are identically defined, even then each backend 
+   is treated differently and results in having their individual default FPGs
+   when default share creation is done using both the backends.
```

Comment:
@imran-ansari  I think we should add below limitations as well. Or can pick the necessary limitations of these:
1. While working in K8S environment it is recommended to use below steps:
If working with dynamic provisioner-
After creating SC and PVC it is recommended to inspect the share to check the status of the share creation.
It is recommended to create a POD only after inspect of share shows status as AVAILABLE. If the user tries to create a POD
before inspect of share staus becomes AVAILABLE, user will see error messages in logs indicating share is in CREATING state.
POD describe may show warning messages indicating Mount has failed as share is still in creating state.

If working with Static provisioner-
After creation of PV and PVC actula share doesnt get created at 3PAR. When user creates POD, dory invokes create share and later 
it invokes mount share commands. Hence user may experiece few minutes of delay for getting POD in running state.
User may also see multiple error messages indication "mount for share is failed as share is still in creating state".
This is because once the PVC status becomes BOUND dory tries to create pod continuously. 



2. There can be maximum of 256 shares which can be created using docker volume plugin. This is because,
share is internally mapped to docker volume. There is a 3PAR limit on number of FPGs per node pair (16 FPGs).
Each FPG can have max 16 stores. There is 1:1 mapping mentained for store and share.

3. Even though 3PAR vfs can have max 4 IPs associated with it, docker user can have max 1 IP associated with VFS.

Reply:
PR updated accordingly. Please check.